mahidhar/segtree.cpp

```diff
@@ -0,0 +1,106 @@
+//This is a generic implementation of the segment tree without lazy propagation
+//This solves range sum query with updates
+//For other problems the implementation of the util class should be changed accordingly.
+
+#include <iostream>
+
+struct sTreeNode{
+	int x;
+};
+
+typedef struct sTreeNode snode;
+
+class util{
+public:
+	//This is the return value if the range is out of bounds
+	snode outValue;
+
+	util(){
+		outValue.x = 0;
+	}
+	snode getLeaf(int a){
+		snode ans;
+		ans.x = a;
+		return ans;
+	}
+	snode combine(const snode &a,const snode &b){
+		snode ans;
+		ans.x = a.x+b.x;
```

Comment:
To make the code readable you should add a space before and after binary operators like +, similar to what you have done with the "="